instrumentor-aop/src/main/java/com/sproutsocial/metrics/InstrumentationDetails.java

```diff
@@ -0,0 +1,60 @@
+package com.sproutsocial.metrics;
+
+import com.google.common.base.Strings;
+import org.aopalliance.intercept.MethodInvocation;
+
+import java.lang.reflect.Method;
+
+/**
+ * Created on 2/29/16.
+ */
+public interface InstrumentationDetails {
+
+    Instrumented getAnnotation(MethodInvocation methodInvocation);
+    String name(Method method, Instrumented annotation);
+
+    class ClassInstrumentation implements InstrumentationDetails {
```

Comment:
Sure -- the idea here is that if you have a `name` attribute on a class-level `@Instrumented`, you could be instrumenting more than one method. In this case, I think ideally you'd want all of its methods instrumented separately. To accomplish this, take whatever `name` was given for the class, and then add on the method name.

This is not a concern in the method case, because you cannot specify multiple methods using a single `@Instrumented` annotation.

Aside: A use case I hadn't considered is where you want all the instrumentation of the class to bundle the stats for all of that class's methods into a single set of metrics. But I think that having more granularity should be the default -- clients can always aggregate later.

Aside aside: Bundling stats for multiple methods under the same `name` is still possible with multiple method-scoped `@Instrumented`s with the same `name` -- if someone really wanted that functionality.